netci.groovy

```diff
@@ -658,35 +658,52 @@ branchList.each { branchName ->
     }
 }
 
-// Generate the build and test versions for Windows_NT.  When full build/run is supported on a platform, those platforms
+// Generate the build and test versions for Windows_NT and Ubuntu.  When full build/run is supported on a platform, those platforms
 // could be removed from above and then added in below.
-def supportedFullCyclePlatforms = ['Windows_NT']
+def supportedFullCyclePlatforms = ['Windows_NT', 'Ubuntu']
 
 branchList.each { branchName ->
     ['Debug', 'Release'].each { configurationGroup ->
-        supportedFullCyclePlatforms.each { osGroup ->
+        supportedFullCyclePlatforms.each { os ->
+            def osGroup = osGroupMap[os]
             def isPR = (branchName == 'pr') 
-            def newJobName = "${osGroup.toLowerCase()}_${configurationGroup.toLowerCase()}"
+            def newJobName = "${os.toLowerCase()}_${configurationGroup.toLowerCase()}"
 
             def newJob = job(getJobName(Utilities.getFullJobName(project, newJobName, isPR), branchName)) {
                 steps {
-                    batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && build.cmd /p:ConfigurationGroup=${configurationGroup} /p:OSGroup=${osGroup}")
-                    batchFile("C:\\Packer\\Packer.exe .\\bin\\build.pack .\\bin")
+                    if (os == 'Windows 10' || os == 'Windows 7' || os == 'Windows_NT') {
+                        batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && build.cmd /p:ConfigurationGroup=${configurationGroup} /p:OSGroup=${osGroup}")
+                        batchFile("C:\\Packer\\Packer.exe .\\bin\\build.pack .\\bin")
+                    }
+                    else if (os == 'OSX') {
+                        shell("HOME=\$WORKSPACE/tempHome ./build.sh /p:ConfigurationGroup=${configurationGroup} /p:TestWithLocalLibraries=true")
+                    }
+                    else {
+                        shell("sudo HOME=\$WORKSPACE/tempHome ./build.sh /p:ConfigurationGroup=${configurationGroup} /p:TestWithLocalLibraries=true")
```

Comment:
TestWithLocalLibraries is sort-of required for lots of projects in x-plat builds. If the native shims are changed in the PR then this needs to be passed, because otherwise we will pull in old shims from NuGet packages. We should be able to have more fine-grained control over what "local libraries" we need to copy, but this is the best we have right now, I think.

Technically, this isn't strictly an x-plat thing, but Windows doesn't have native components that change very often, so we don't hit the issue there.